processing/src/main/java/io/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -79,7 +81,108 @@ public GroupByParallelQueryRunner(
   }
 
   @Override
-  public Sequence<T> run(final Query<T> queryParam, final Map<String, Object> responseContext)
+  public Sequence<T> run(final Query<T> query, final Map<String, Object> responseContext)
+  {
+    boolean isSingleThreaded = query.getContextValue(CTX_KEY_IS_SINGLE_THREADED, configSupplier.get().isSingleThreaded());
+    if (isSingleThreaded) {
+      return runSingleThreaded(query, responseContext);
+    } else {
+      return runParallel(query, responseContext);
+    }
+  }
+
+  private Sequence<T> runSingleThreaded(final Query<T> queryParam, final Map<String, Object> responseContext)
+  {
+    return new ConcatQueryRunner<>(
+        Sequences.map(
+            Sequences.simple(queryables),
+            new Function<QueryRunner<T>, QueryRunner<T>>()
+            {
+              @Override
+              public QueryRunner<T> apply(final QueryRunner<T> input)
+              {
+                return new QueryRunner<T>()
+                {
+                  @Override
+                  public Sequence<T> run(final Query<T> query, final Map<String, Object> responseContext)
+                  {
+                    final Pair<IncrementalIndex, Accumulator<IncrementalIndex, T>> indexAccumulatorPair = GroupByQueryHelper
```

Comment:
@xvrl they are quite similar but just a bit different, i just moved the code around but now that I'm looking closely at the code, i think it can be refactored and improved a bit ... it seems in singleThreaded case one index is being created per runner from input set of runners while parallel one creates only one index for all of them... same should be doable for single threaded case also.
I'm making the changes, but they look a bit intrusive and unrelated to this PR. So, I propose that I will submit another PR soon with refactoring and possible improvement.

let us get this one merged as I have some people waiting for this feature to get delivered.